### PR TITLE
feat: sort repositories by columns from the api

### LINKED ISSUE #627: feature: Filter by column type on Repository screen
### Type of feature

Select the type of feature request, the lowercase should also be the PR prefix.

- [x] 🍕 Feature
- [ ] 🐛 Bug Fix
- [ ] 📝 Documentation Update
- [ ] 🎨 Style
- [ ] 🧑‍💻 Code Refactor
- [ ] 🔥 Performance Improvements
- [ ] ✅ Test
- [ ] 🤖 Build
- [ ] 🔁 CI
- [ ] 📦 Chore (Release)
- [ ] ⏩ Revert

### Current behavior

Is your feature request related to a problem? Please describe.

When viewing the Repository page, it has been requesting users, the ability to filter by activity, pr count, etc.  

### Suggested solution

When the user clicks the column the data should sort ascending or descending. 

Screen Shot 2022-11-30 at 4 27 14 PM

### Code of Conduct

By submitting this issue, you agree to follow our [Code of Conduct](https://docs.opensauced.pizza/contributing/code-of-conduct/).

- [ ] I agree to follow this project's Code of Conduct

### Contributing Docs

If you plan on contributing code please read - [Contribution Guide](https://docs.opensauced.pizza/contributing/introduction-to-contributing/)

- [ ]  I agree to follow this project's Contribution Docs

### PR DESCRIPTION
**Pull request by Deadreyo**
## What type of PR is this? (check all applicable)

- [x] 🍕 Feature
- [ ] 🐛 Bug Fix
- [ ] 📝 Documentation Update
- [ ] 🎨 Style
- [ ] 🧑‍💻 Code Refactor
- [ ] 🔥 Performance Improvements
- [ ] ✅ Test
- [ ] 🤖 Build
- [ ] 🔁 CI
- [ ] 📦 Chore (Release)
- [ ] ⏩ Revert

## Description

- Allows clicking on column names to sort repositories by columns, by modifying the request sent to the api.
- Adds `sortBy` data to the global state.
- Takes some parts from #662 by @TAKANOME-DEV 

## Related Tickets & Documents

closes #627 
## Added tests?

- [ ] 👍 yes
- [ ] 🙅 no, because they aren't needed
- [ ] 🙋 no, because I need help

## Added to documentation?

- [ ] 📜 README.md
- [ ] 📓 docs.opensauced.pizza
- [ ] 🍕 dev.to/opensauced
- [ ] 📕 storybook
- [ ] 🙅 no documentation needed

### CODE REVIEW

**Comment by OgDev-01** [comment]
filter gives error
image

**Comment by Deadreyo** [comment]
> filter gives error image

Yeah that's related to non-existing queries that I'm using as a placeholder

**Comment by brandonroberts** [comment]
> Backend issues:
> 
> 1. `orderBy` in the `{topic}/repos` doesn't seem to work in the following example link:
>    https://beta.gs-api.opensauced.pizza/v1/javascript/repos?page=1&limit=10&range=30&filter=minimum-5-contributors&orderBy=stars
> 
> Changing `orderBy` in the above link doesn't seem to change anything.
> 
> 2. Need to know what each column corresponds to in the `orderBy` options, and some columns don't seem to have corresponding options as well.

This is expected as the minimum contributors sorts by the `repo_recent_contribution_count` in the data. When you have no filters applied, the stars sorting works as expected. The other filters have custom sorted applied before sorting by stars

**Comment by Deadreyo** [comment]
> This is expected as the minimum contributors sorts by the `repo_recent_contribution_count` in the data. When you have no filters applied, the stars sorting works as expected. The other filters have custom sorted applied before sorting by stars

So what you mean is column sorting isn't intended to be used with filters, if I understand correctly?

**Comment by Deadreyo** [comment]
So what's the best path of action that should be taken here?

**Comment by brandonroberts** [comment]
I think putting the sort options in is still needed. The sorting is not global though, so it should be confined to the repositories tab. The `lib/store` is for global shared state only. It also should not be importing any downstream components/code.

**Comment by Deadreyo** [comment]
@OgDev-01 @brandonroberts @0-vortex @bdougie 
Ready for review.

Been blocked for a while due to the columns not having corresponding query options in the backend. So now updates on that would be in a new issue.
Currently only "Repositories" can be sorted since it has a corresponding query.

**Comment by bdougie** [comment]
> Been blocked for a while due to the columns not having corresponding query options in the backend.

Have you opened an issue in https://github.com/open-sauced/ops-backend?

You have two options. 

1. Open the issue
2. Open the PR.

**Comment by Deadreyo** [comment]
@bdougie opened
https://github.com/open-sauced/ops-backend/issues/34

File: components/organisms/Repositories/repositories.tsx [hunk 1]
```diff
@@ -79,10 +91,32 @@ const Repositories = ({ repositories }: RepositoriesProps): JSX.Element => {
     }
   };
 
+  const handleSortByColumn = (column: SortOptions) => {
+    // debugger
```
Comment (NsdHSO) [inline components/organisms/Repositories/repositories.tsx:95]: Could you please remove these  lines of comment

**Comment by brandonroberts** [comment]
Closing for now. Will reopen when backend API decision is made

**Pull request closed without merging**